src/QsCompiler/Transformations/ClassicallyControlledTransformation.cs

```diff
@@ -4,615 +4,472 @@
-                    // In this context, that is a call back to the original callable from the newly generated operation
-                    if (sym is Identifier.GlobalCallable callable && _super._OldName.Equals(callable.Item))
+                    if (expression.Expression is ExpressionKind.EQ eq)
                     {
-                        // Setting this flag will prevent the rerouting logic from processing the resolved type of the recursive identifier expression.
-                        // This is necessary because we don't want any type parameters from the original callable from being rerouted to the new generated
-                        // operation's type parameters in the definition of the identifier.
-                        _super._IsRecursiveIdentifier = true;
+                        if (eq.Item1.Expression is ExpressionKind.ResultLiteral exp1)
+                        {
+                            return (true, exp1.Item, eq.Item2);
+                        }
+                        else if (eq.Item2.Expression is ExpressionKind.ResultLiteral exp2)
+                        {
+                            return (true, exp2.Item, eq.Item1);
+                        }
                     }
-                    return rtrn;
-                }
-            }
 
-            private class UpdateGeneratedOpExpressionType : ExpressionTypeTransformation<UpdateGeneratedOpExpression>
-            {
-                private UpdateGeneratedOpTransformation _super;
+                    return (false, null, null);
+                }
 
-                public UpdateGeneratedOpExpressionType(UpdateGeneratedOpTransformation super, UpdateGeneratedOpExpression expr) : base(expr) { _super = super; }
+                #endregion
 
-                public override ResolvedTypeKind onTypeParameter(QsTypeParameter tp)
+                public override QsScope Transform(QsScope scope)
                 {
-                    // Reroute a type parameter's origin to the newly generated operation
-                    if (!_super._IsRecursiveIdentifier && _super._OldName.Equals(tp.Origin))
+                    var parentSymbols = this.onLocalDeclarations(scope.KnownSymbols);
+                    var statements = new List<QsStatement>();
+
+                    foreach (var statement in scope.Statements)
                     {
-                        tp = new QsTypeParameter(_super._NewName, tp.TypeName, tp.Range);
+                        if (statement.Statement is QsStatementKind.QsConditionalStatement)
+                        {
+                            var stm = ReshapeConditional(statement);
+                            stm = this.onStatement(stm);
+
+                            var (isCondition, cond, conditionScope, defaultScope) = IsConditionWtihSingleBlock(stm);
+
+                            if (isCondition)
+                            {
+                                /*ToDo: this could be a separate function.*/
```

Comment:
Ok, so why isn't it a separate function? I.e. why is it left as a todo?

Reply:
This will be addressed in the Classical Control Rewrite feature.